src/components/NoteListSection/NoteListSection.jsx

```diff
@@ -1,13 +1,111 @@
-import React from "react";
+import React, { useState } from "react";
 import ReactQuill from "react-quill";
+import { IconButton } from "components";
+import { AiOutlineFilter } from "react-icons/ai";
+import { RiSortDesc } from "react-icons/ri";
 import "react-quill/dist/quill.bubble.css";
+import { getFilteredList } from "utils/filters";
+import { getTags } from "utils/getTags";
+
+const priorities = ["High", "Medium", "Low"];
+
+const NoteListSection = ({
+  notesList,
+  selectedNote,
+  setSelectedNote,
+  filterState,
+  filterDispatch,
+}) => {
+  const filteredList = getFilteredList(notesList, filterState);
+
+  const tags = getTags(notesList);
 
-const NoteListSection = ({ notesList, selectedNote, setSelectedNote }) => {
   return (
     <section className="note-list-section">
       {/* add options for sort and filter here */}
+      <section className="filter-section border-bottom">
+        <p>{filteredList.length} Notes</p>
+        <div className="filter-options">
+          <button
+            className="btn clear-filter-btn"
+            onClick={() => filterDispatch({ type: "CLEAR_FILTERS" })}
+          >
+            Clear Filters
+          </button>
+          <div className="dropdown-wrapper dropdown-wrapper-filter">
+            <IconButton icon={<AiOutlineFilter />} />
+            <div className="dropdown filter">
+              <section>
+                <span className="text-center">Filter by Priority</span>
+                {priorities.map((priority) => {
+                  return (
+                    <button
+                      className={`btn ${
+                        filterState.priority == priority.toLowerCase()
+                          ? "btn-selected"
+                          : ""
+                      }`}
+                      onClick={() =>
+                        filterDispatch({
+                          type: "FILTER_BY_PRIORITY",
+                          payload: priority.toLowerCase(),
+                        })
+                      }
+                    >
+                      {priority}
+                    </button>
+                  );
+                })}
```

Comment:
You can also write its outside of your jsx.

let priortyJsx = your code

in jsx 

 <div>{priorityJsx}</div>